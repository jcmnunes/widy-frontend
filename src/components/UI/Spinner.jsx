import styled, { keyframes } from 'styled-components';

const getColors = props => {
  switch (props.intent) {
    case 'primary':
    case 'success':
    case 'warning':
    case 'error':
      return {
        border: '#FFFFFF32',
        accent: '#FFF',
      };
    default:
      return {
        border: `${props.theme.neutral200}`,
        accent: `${props.theme.neutral300}`,
      };
  }
};

const spin = keyframes`
  100% { 
    transform: rotate(360deg); 
  } 
`;

const Spinner = styled.span`
  display: inline-block;
  box-sizing: border-box;
  width: ${props => (props.size === 'large' ? '24px' : '18px')};
  height: ${props => (props.size === 'large' ? '24px' : '18px')};
  border-radius: 100%;
  border: ${props => `3px solid ${getColors(props).border}`};
  border-top-color: ${props => getColors(props).accent};
  animation: ${spin} 500ms infinite linear;
`;

export default Spinner;
