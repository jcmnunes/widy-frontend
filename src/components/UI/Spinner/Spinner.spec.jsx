import React from 'react';
import Spinner from '.';

const TestRenderer = require('react-test-renderer');

describe('UI Components', () => {
  describe('RoundButton', () => {
    const props = {};

    it('should render as expected, without crashing', () => {
      const tree = TestRenderer.create(<Spinner {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should apply the correct styles for a large spinner', () => {
      const newProps = {
        ...props,
        size: 'large',
      };
      const tree = TestRenderer.create(<Spinner {...newProps} />).toJSON();
      expect(tree).toHaveStyleRule('width', '24px');
      expect(tree).toHaveStyleRule('height', '24px');
    });

    it('should apply the correct styles for a non neutral (or secondary) spinner', () => {
      const newProps = {
        ...props,
        intent: 'primary',
      };
      const tree = TestRenderer.create(<Spinner {...newProps} />).toJSON();
      expect(tree).toHaveStyleRule('border', '3px solid #FFFFFF32');
      expect(tree).toHaveStyleRule('border-top-color', '#FFF');
    });
  });
});
