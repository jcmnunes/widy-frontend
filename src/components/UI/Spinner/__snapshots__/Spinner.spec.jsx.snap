// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Components RoundButton should render as expected, without crashing 1`] = `
.c0 {
  display: inline-block;
  box-sizing: border-box;
  width: 18px;
  height: 18px;
  border-radius: 100%;
  border: 3px solid undefined;
  border-top-color: undefined;
  -webkit-animation: cZxgpV 500ms infinite linear;
  animation: cZxgpV 500ms infinite linear;
}

<span
  className="c0"
  size="small"
/>
`;
